ocs_ci/helpers/managed_services.py

```diff
@@ -0,0 +1,191 @@
+    log.info(f"Verified that the size of each OSD is {osd_size}Ti")
+
+    # Verify OSD running worker nodes count
+    osd_nodes = get_osd_running_nodes()
+    assert len(osd_nodes) == size_map[size]["instance_count"], (
+        f"Worker node instance count is not as expected. Actual instance count is {len(osd_nodes)}. "
+        f"Expected {size_map[size]['instance_count']}. List of worker nodes : {osd_nodes}"
+    )
+    log.info("Verified the number of worker nodes where OSD is running.")
+
+    # Verify worker node instance type
+    worker_node_names = get_worker_nodes()
+    worker_nodes = get_node_objs(worker_node_names)
+    for node_obj in worker_nodes:
+        assert (
+            node_obj.get("metadata")
+            .get("metadata")
+            .get("labels")
+            .get("beta.kubernetes.io/instance-type")
+            == instance_type
+        ), f"Instance type of the worker node {node_obj.name} is not {instance_type}"
+    log.info(f"Verified that the instance type of wokeer nodes is {instance_type}")
+
+    # Verify OSD count
+    osd_count = get_osd_count()
+    assert (
+        osd_count == size_map[size]["osd_count"]
+    ), f"OSD count is not as expected. Actual:{osd_count}. Expected:{size_map[size]['osd_count']}"
+    log.info(f"Verified that the OSD count is {size_map[size]['osd_count']}")
+
+    # Verify OSD cpu
+    osd_cpu_limit = "2"
+    osd_cpu_request = "1"
+    osd_pods = get_osd_pods()
+    log.info("Verifying OSD cpu")
+    for osd_pod in osd_pods:
+        for container in osd_pod.data["spec"]["containers"]:
+            if container["name"] == "osd":
+                assert container["resources"]["limits"]["cpu"] == osd_cpu_limit, (
+                    f"OSD pod {osd_pod.name} container osd doesn't have cpu limit {osd_cpu_limit}. "
+                    f"Limit is {container['resources']['limits']['cpu']}"
+                )
+                assert container["resources"]["requests"]["cpu"] == osd_cpu_request, (
+                    f"OSD pod {osd_pod.name} container osd doesn't have cpu request {osd_cpu_request}. "
+                    f"Request is {container['resources']['requests']['cpu']}"
+                )
+    log.info("Verified OSD CPU")
+
+    # Verify machine pools
+    cmd = f"rosa list machinepool --cluster={config.ENV_DATA['cluster_name']} -o yaml"
+    out = run_cmd(cmd)
+    machine_pool_info_list = yaml.safe_load(out)
+    assert (
+        len(machine_pool_info_list) == 2
+    ), f"Number of machinepools is not 2. Machinepools details: {machine_pool_info_list}"
+
+    machine_pool_ids = []
+    ceph_osd_nodepool_info = None
+    for machine_pool_info in machine_pool_info_list:
+        machine_pool_ids.append(machine_pool_info["id"])
+        assert (
+            machine_pool_info["instance_type"] == instance_type
+        ), f"Instance type of machinepool {machine_pool_info['id']} is {machine_pool_info['instance_type']}"
```

Comment:
I would also mention the expected `instance_type` in the assert message.

Reply:
This will be added in the follow-up PR.